kubernetes/cache/kube_cache.go

```diff
@@ -1464,15 +1464,27 @@ func (c *kubeCache) GetTelemetries(namespace, labelSelector string) ([]*v1alpha1
 	return retTelemetries, nil
 }
 
+func (c *kubeCache) IsK8sGatewayListerInit(namespace string) bool {
+	// K8s GW has several cache listers, those can be namespace or cluster based
+	// if one of them is not initialized, then all others are not init as well
+	// this can happen when CRDs are created after Kiali start
+	if c.getCacheLister(namespace).k8sgatewayLister == nil {
+		log.Info("K8s Gateway API CRDs are installed, Kiali needs to be restarted to apply.")
+	}
+	return c.getCacheLister(namespace).k8sgatewayLister != nil
```

Comment:
```suggestion
	if c.getCacheLister(namespace).k8sgatewayLister == nil {
		log.Info("K8s Gateway API CRDs are installed, Kiali needs to be restarted to apply.")
		return false
	}
	return true
```

No need to do the lookup/test twice.